src/app/child-dev-project/attendance/model/activity-attendance.ts

```diff
@@ -0,0 +1,150 @@
+import { Note } from "../../notes/model/note";
+import {
+  AttendanceCounting,
+  AttendanceStatus,
+  DEFAULT_ATTENDANCE_TYPES,
+} from "./attendance-status";
+import { Entity } from "../../../core/entity/entity";
+import { RecurringActivity } from "./recurring-activity";
+
+export class ActivityAttendance extends Entity {
+  static create(from: Date, events: Note[] = []) {
+    const instance = new ActivityAttendance();
+    instance.periodFrom = from;
+    instance.events = events;
+    return instance;
+  }
+
+  periodFrom: Date;
+  periodTo: Date;
+
+  events: Note[] = [];
+
+  activity: RecurringActivity;
+
+  /**
+   * childId to be used as default when getting statistics through the instance's methods (e.g. getEventsPresent)
+   *
+   * if `focusedChild` is set and a method like `getEventsPresent()` is called without parameter
+   * the focusedChild is used as parameter implicitly to simplify calling these methods in templates.
+   */
+  focusedChild: string;
```

Comment:
agreed. This was a workaround because I could not pass a parameter to a function for EntitySubrecordComponent columns. We have added this with the `valueFunction` definition. I'll simplify the data model here, making use of that instead.